qemu/qemu.go

```diff
@@ -26,14 +26,876 @@ package qemu
 
 import (
 	"bytes"
+	"fmt"
 	"os"
 	"os/exec"
+	"strings"
 
 	"context"
 )
 
-// LaunchQemu can be used to launch a new qemu instance by invoking the
-// qemu-system-x86_64 binary.
+// Machine describes the machine type qemu will emulate.
+type Machine struct {
+	// Type is the machine type to be used by qemu.
+	Type string
+
+	// Acceleration are the machine acceleration options to be used by qemu.
+	Acceleration string
+}
+
+// Device is the qemu device interface.
+type Device interface {
+	Valid() bool
+}
+
+// DeviceDriver is the Device driver string.
+type DeviceDriver string
+
+const (
+	// NVDIMM is the Non Volatile DIMM device driver.
+	NVDIMM DeviceDriver = "nvdimm"
+
+	// Virtio9P is the 9pfs device driver.
+	Virtio9P = "virtio-9p-pci"
+
+	// VirtioNet is the virt-io networking device driver.
+	VirtioNet = "virtio-net"
+
+	// VirtioSerial is the serial device driver.
+	VirtioSerial = "virtio-serial-pci"
+
+	// Console is the console device driver.
+	Console = "virtconsole"
```

Comment:
We should probably switch to the newer variant of the same
   -drive file="$IMAGE",if=none,id="$DRIVE",aio=threads,format=qcow2 \
   -device virtio-blk,drive="$DRIVER",scsi=off,config-wce=off \